### DISCUSSION TITLE
long polling issues

### DISCUSSION THREAD

**Discussion by kolyande** [reo7sp/tgbot-cpp#251] [Q&A] [answered]
Hello!

While using BoostHttpOnlySslClient when timeout much greater than 10 seconds is set for TgLongPoll (50 sec, 100 sec) it leads to std::exception throwing (from here https://github.com/reo7sp/tgbot-cpp/blob/master/src/net/BoostHttpOnlySslClient.cpp#L74) in few seconds after TgLongPoll::start() method call.
What is the reason of that - am I doing something wrong?

And one more - in TgLongPoll::start() method Api::getUpdates() is called twice (here https://github.com/reo7sp/tgbot-cpp/blob/master/src/net/TgLongPoll.cpp#L25 and here https://github.com/reo7sp/tgbot-cpp/blob/master/src/net/TgLongPoll.cpp#L36) both times with the same predefined timeout value, so result is that real long polling period is twice bigger than expected. Or this is how it should work?

Thanks.

**Comment by we-lu** [comment]
Hi!

Thanks for your message.

1. No, that's not supposed to happen. The exception is thrown if the timeout is > 20 seconds. This is because the read timeout in `BoostHttpOnlySslClient` is hard coded to 20 instead of the value of `TgLongPoll::_timeout`. The same is true for the CurlHttpClient, where the `CURLOPT_TIMEOUT` is set to 25 instead of `TgLongPoll::_timeout`. Currently, I don't have time to fix this. You can temporarily change these two values manually. I'll commit a fix asap if it's not already done by someone else in the meantime.

2. No, this is also not correct. I can't fix this quickly, due to the current structure of `TgLongPoll::start()`. You can do it quick and dirty if you set `_timeout = _timeout / 2`.

I'll keep you updated.

**Comment by we-lu** [comment]
Hello!

The issues should now be fixed with the newest commit.

**Comment by reo7sp** [comment, marked as answer]
Both old issues are resolved. Current `TgLongPoll::start()` performs one `getUpdates()` request per iteration, and the HTTP timeout is derived from the configured long-poll timeout. In v1.12 use `startLoop(errorHandler)` for continuous polling.
